src/FilterList.js

```diff
@@ -2,16 +2,84 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import {ResultPropTypes} from './ResultPropTypes'
 
-const FilterList = ({filteredResults, resultsMessage, ResultsHeaderClass, ResultElementClass}) => {
-  const filteredElements = filteredResults.map((result) => result.element)
-
-  return (
-    <div>
-      <h4>{resultsMessage}</h4>
-      { filteredElements.length && ResultsHeaderClass && <ResultsHeaderClass /> }
-      { filteredElements.map((element, index) => <div key={index}><ResultElementClass {...element}/></div>) }
-    </div>
+function dynamicSort(property, sortState) {
+  const sortOrder = sortState ?  -1 : 1
+  return function (a,b) {
+    let result = (a[property] < b[property]) ? -1 : (a[property] > b[property]) ? 1 : 0
+    return result * sortOrder
+  }
+}
+
+function titleCase(str) {
+  const splitStr = str.toLowerCase().split(` `).map((string, index)=>
+    index === 0 ?
+      string.charAt(0).toLowerCase() + string.substring(1) :
+      string.charAt(0).toUpperCase() + string.substring(1)    
   )
+  return splitStr.join(``)
+}
+
+class FilterList extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      sortTitle: `species`,
```

Comment:
Initialize the `sortTitle` to e.g. the first field of the header, don’t assume any fields are present.